src/Components/Server/src/Circuits/RemoteRenderer.cs

```diff
@@ -101,13 +100,32 @@ protected override void HandleException(Exception exception)
         /// <inheritdoc />
         protected override void Dispose(bool disposing)
         {
+            _disposing = true;
             base.Dispose(true);
-            _rendererRegistry.TryRemove(_id);
+            while (PendingRenderBatches.TryDequeue(out var entry))
+            {
+                try
+                {
+                    entry.CompletionSource.TrySetCanceled();
+                }
+                catch
+                {
+                    // Swallow the exception as we are disposing
+                    // the renderer.
```

Comment:
Should we just be swallowing `TaskCancelledException`, or is there reason to believe we need to suppress all other possible failures here?

Reply:
Turns out I was being over zealous. I don't need the try catch. At this point we are already disposing, so I just want to make sure that things waiting on that TCS get released. (and deal with the cancellation if necessary).

Reply:
No need for the trycatch here